src/Routes/Bot.js

```diff
@@ -0,0 +1,83 @@
+const BaseRoute = require('../Structure/BaseRoute');
+const axios = require('axios');
+const handleError = require('../Util/handleError');
+
+class BotRoute extends BaseRoute {
+    constructor(client, db) {
+        super('/bot');
+        this.router = require('express').Router();
+        this.client = client;
+        this.db = db;
+        this.routes();
+    }
+
+    async postBot(req, res) {
+        const responses = [];
+        return await this.db.select().from('lists').whereNot({ add_bot: null, add_bot_key: null }).then(lists => {
+            for (var list of lists) {
+                if (req.body[list.name] == 'on') {
+                    return axios.post(list.add_bot, req.body, {
+                        headers: { Authorization: list.add_bot_key }
+                    }).then(resp => {
+                        responses.push({ ...resp.data, error: false, name: list.name });
+                        return responses;
+                    }).catch(err => {
+                        responses.push({ ...err.response.data, error: true, name: list.name });
+                        return responses;
+                    });
+                }
+                delete req.body[list.name];
+            }
+        }).catch((e) => {
+            handleError(this.db, req, res, e.stack);
+        });
+    }
+
+    routes() {
+        this.router.get('/add', this.requiresAuth.bind(this), (req, res) => {
+            res.render('bot/add');
+        });
+
+
+        this.router.post('/add', async (req, res) => {
```

Comment:
Should this endpoint require authentiction?

Reply:
You can only get to it from the router.get version of /add which does require authentication. Unless I'm wrong and this is a risk?